### PR TITLE
Create stub for isogram exercise

### PR DESCRIPTION
**Pull request by kytrinyx**
This stub is based on the hello world test, so let me know what I need to tweak in the template in order to make it idiomatic.

One thing we haven't yet talked about is the student experience around test failures.

In some language tracks we've tried to use the mechanisms available in the language and tooling to try to simulate Test-Driven Development, either by
- using a skipping mechanism so that all but the first test is skipped, and then having the student remove or uncomment the skipping mechanism one test at a time
- running tests in the order that they are defined in the file and "failing fast" so that the tests after the first failure are not executed/reported.

Right now if I implement the stub solution to hardcode a return value of `true` I get the following failure:

<details>
<summary>Isogram test failure</summary>

```text
Compiled 'tests.chpl' successfully
Ran test_empty_string() OK
Ran test_isogram_with_only_lower_case_characters() OK
Ran test_word_with_one_duplicated_character() FAIL
Ran test_word_with_one_duplicated_character_from_the_end_of_the_alphabet() FAIL
Ran test_longest_reported_english_isogram() OK
Ran test_word_with_duplicated_character_in_mixed_case() FAIL
Ran test_word_with_duplicated_character_in_mixed_case_lowercase_first() FAIL
Ran test_hypothetical_isogrammic_word_with_hyphen() OK
Ran test_hypothetical_word_with_duplicated_character_following_hyphen() FAIL
Ran test_isogram_with_duplicated_hyphen() OK
Ran test_made_up_name_that_is_an_isogram() OK
Ran test_duplicated_character_in_the_middle() FAIL
Ran test_same_first_and_last_characters() FAIL
Ran test_word_with_duplicated_character_and_with_two_hyphens() FAIL

======================================================================
FAIL tests.chpl: test_word_with_one_duplicated_character()
----------------------------------------------------------------------
AssertionError: assert failed - 'true' != 'false'

======================================================================
FAIL tests.chpl: test_word_with_one_duplicated_character_from_the_end_of_the_alphabet()
----------------------------------------------------------------------
AssertionError: assert failed - 'true' != 'false'

======================================================================
FAIL tests.chpl: test_word_with_duplicated_character_in_mixed_case()
----------------------------------------------------------------------
AssertionError: assert failed - 'true' != 'false'

======================================================================
FAIL tests.chpl: test_word_with_duplicated_character_in_mixed_case_lowercase_first()
----------------------------------------------------------------------
AssertionError: assert failed - 'true' != 'false'

======================================================================
FAIL tests.chpl: test_hypothetical_word_with_duplicated_character_following_hyphen()
----------------------------------------------------------------------
AssertionError: assert failed - 'true' != 'false'

======================================================================
FAIL tests.chpl: test_duplicated_character_in_the_middle()
----------------------------------------------------------------------
AssertionError: assert failed - 'true' != 'false'

======================================================================
FAIL tests.chpl: test_same_first_and_last_characters()
----------------------------------------------------------------------
AssertionError: assert failed - 'true' != 'false'

======================================================================
FAIL tests.chpl: test_word_with_duplicated_character_and_with_two_hyphens()
----------------------------------------------------------------------
AssertionError: assert failed - 'true' != 'false'

----------------------------------------------------------------------
Ran 14 tests in 7.71508 seconds

FAILED (passed = 6 failures = 8 )
```

</details>

This isn't the end of the world, but in terms of pedagogy it's nice when the next step is clearly indicated by having a failure that is adjacent to the current implementation, if that makes sense.

### CODE REVIEW

**Comment by lucaferranti** [comment]
> This isn't the end of the world, but in terms of pedagogy it's nice when the next step is clearly indicated by having a failure that is adjacent to the current implementation, if that makes sense.

I'm not sure I understand what you mean. In general, I agree error messages could be better in Chapel. The good news is, we are not bound to use `UnitTest` if we want more control on the printed output, it's not too difficult to setup one's own test framework in Chapel. Can you give a MWE of the output you'd expect to see?

**Comment by kytrinyx** [comment]
Ah, I didn't mean that the error messages are bad, only that because we've pre-written the entire test suite we get 30 errors all at once instead of getting one error at a time.

When people are doing TDD on their own, they generally just write "the next test", and therefore are able to focus in on just one little thing at a time, but when there are a bunch of errors up front it's hard to know what to focus on next in terms of solving the exercise.

**Comment by lucaferranti** [comment]
sorry I didn't read carefully your message 😅 

> using a skipping mechanism so that all but the first test is skipped, and then having the student remove or uncomment the skipping mechanism one test at a time

I think this could be implemented using Chapel `skipIf` and `dependsOn`. I can try drafting a quick prototype.
How would this work if students work from the web editor though? I think in that case they couldn't edit the tests themselves.

> running tests in the order that they are defined in the file and "failing fast" so that the tests after the first failure are not executed/reported.

yes this should be possible too by setting chapel to halt on error, I'll quickly check with a small demo.

I guess pedagogically this gives the student a more guided workflow, forcing them to focus on one small step at the time, which can be good. A possible downside of fast failures is that if e.g. while working on test 5 I accidentally break test 1, then I don't see if it affects also tests 2,3,4, wouldn't this slow down debugging?

**Comment by kytrinyx** [comment]
> How would this work if students work from the web editor though? I think in that case they couldn't edit the tests themselves.

That is a fantastic question. ~I think for the webeditor we remove the skips ahead of time, maybe. @ErikSchierboom do you happen to know off the top of your head what the usual approach is to this?~ Erik: we got an answer to this.

**Comment by kytrinyx** [comment]
> I guess pedagogically this gives the student a more guided workflow, forcing them to focus on one small step at the time, which can be good.

Yeah, exactly. That's the main idea.

> A possible downside of fast failures is that if e.g. while working on test 5 I accidentally break test 1, then I don't see if it affects also tests 2,3,4, wouldn't this slow down debugging?

Yeah, fail-fast will slow down debugging. Skipping/unskipping will not, because all the previously "solved" tests will still fail if you make a change that breaks earlier tests, which I think is a better experience.

**Comment by kytrinyx** [comment]
> I think this could be implemented using Chapel skipIf and dependsOn. I can try drafting a quick prototype.

That would be great.

(I should have started this discussion in an issue rather than a random PR /facepalm -- I'll open an issue about it now)

File: config.json [hunk 1]
```diff
@@ -54,7 +54,6 @@
         "name": "Leap",
         "practices": [],
         "prerequisites": [],
-<<<<<<< HEAD
```
Comment (kytrinyx) [inline config.json]: Oh, jeez. I'm so sorry!

**Comment by kytrinyx** [comment]
This looks great. I can't hit the approve button since I opened the PR, but I totally do approve :)

**Review by lucaferranti — APPROVED** [review]

**Pull request merged**